### PR TITLE
(#119) Replace CCA/cca term with 'islet' in core APIs and in code

### PR DESCRIPTION
**Pull request by gapisback**
This commit renames stuff like 'cca', 'CCA' in various Islet APIs and interfaces, and code-dir / artifact-names to become
'islet' or ISLET, as appropriate. This naming is inline with what was done for Keystone-based simple-app.

There are just a couple of remaining instances of cca, only used for shim-produced 'bins', in the driver shell script, but these are harmless:

```
 743     elif [ "${SampleAppName}" = "simple_app_under_islet" ]; then
 744         setup_cca_emulated_islet_shim_bins

 786 function setup_cca_emulated_islet_shim_bins() {
 787     run_cmd
 788     run_pushd "${PROV_DIR}"
 789
 790     run_cmd cp -p policy_cert_file.bin cca_emulated_islet_key_cert.bin

1221     elif [ "${SampleAppName}" = "simple_app_under_islet" ]; then
1222         subject_arg="--cert_subject"
1223         key_subject_file="cca_emulated_islet_key_cert.bin"

```

### CODE REVIEW

**Review by jlmucb — APPROVED** [review]

File: sample_apps/simple_app_under_islet/temporary_instructions.txt [hunk 1]
```diff
@@ -116,61 +111,71 @@ make dummy
...
       --policy_store_file=policy_store --measurement_file="example_app.measurement" &
 
-sleep 5
+sleep 10
 
-cd $CCA_EXAMPLE_DIR
+cd $ISLET_EXAMPLE_DIR
 
-$CCA_EXAMPLE_DIR/cca_example_app.exe --print_all=true --operation=run-app-as-client --data_dir=./app1_data/ \
+$ISLET_EXAMPLE_DIR/islet_example_app.exe --print_all=true --operation=run-app-as-client --data_dir=./app1_data/ \
```
Comment (gapisback) [inline sample_apps/simple_app_under_islet/temporary_instructions.txt:179]: This script now mostly works, but this very last step fails with a SSL-error:

```
+ /home/agurajada/Code/certifier-framework-for-confidential-computing/sample_apps/simple_app_under_islet/islet_example_app.exe --data_dir=./app1_data/ --operation=run-app-as-client --policy_store_file=policy_store --print_all=true
Running App as client
init_client_ssl: ssl_connect failed, ret=-1, err=1: SSL_ERROR_SSL: generic ssl error
Can't init client app
server_channel_accept_and_auth: Can't SSL_accept connection, res=-1
Accept error: SSL routines
SSL_ERROR_ZERO_RETURN: zero return error
```

It's unclear why this fails.

The very same steps implemented by `run_example.sh` work just fine.

I tried debugging this somewhat but could not find the problem. Will leave it for now as the driver bash script is working just fine.

**Comment by gapisback** [comment]
HI, @bitboom and @jinpark -- The changes are ready, as requested. As you had asked for this renaming, I wanted to give you a chance to review this work. (Please see my email w/ details.)

@jlmucb has passed it earlier ... based on early draft-PR.

File: src/islet/islet_shim.cc [hunk 2]
```diff
@@ -26,22 +26,22 @@ using namespace certifier::utilities;
 static const char CLAIM_TITLE_USER_DATA[] = "User data";
 static const char CLAIM_TITLE_RIM[] = "Realm initial measurement";
 
-bool cca_Init(const int cert_size, byte *cert) {
+bool islet_Init(const int cert_size, byte *cert) {
   return true;
 }
 
-bool cca_Attest(const int what_to_say_size, byte* what_to_say,
+bool islet_Attest(const int what_to_say_size, byte* what_to_say,
```
Comment (gapisback) [inline src/islet/islet_shim.cc:33]: There is a little bit of very close names with this change. 

The new name `islet_Attest()` is very close to the actual API, `islet_attest()`; just the upper-case `A` is the diff.

Same issue applies to other interface functions.

But I can't do much about it w/o reworking the naming scheme all over.

If you can live with this naming closeness, ... I suggest we leave it alone for now.

File: src/islet/islet_shim.cc [hunk 3]
```diff
@@ -69,33 +69,33 @@ bool cca_Verify(const int what_to_say_size, byte* what_to_say,
...
+  if (memcmp(islet_what_to_say_returned, islet_what_to_say_expected, len) != 0)
     return false;
 
   rv = islet_parse(CLAIM_TITLE_RIM, claims, claims_len, measurement_out,
                    measurement_out_size);
   return rv == ISLET_SUCCESS;
 }
 
-bool cca_Seal(int in_size, byte* in, int* size_out, byte* out) {
+bool islet_Seal(int in_size, byte* in, int* size_out, byte* out) {
```
Comment (gapisback) [inline src/islet/islet_shim.cc:93]: Same 'issue': `islet_Seal()` is too close to lower level API `islet_seal()`.

Comment (gapisback) [reply src/islet/islet_shim.cc:93]: One option for you to consider for the future is maybe to name the lower-level APIs as one of:

- `islet_seal_()`
- `_islet_seal()`
- `islet__seal()`

None of these are really attractive. I'll let @bitboom and @jinbum think about this separately.

**Review by bitboom — APPROVED** [review]
Really thanks for quickly update 👍

**Pull request merged**